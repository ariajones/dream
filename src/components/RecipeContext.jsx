import React, { useState, createContext, useEffect } from 'react'
import firebase from 'firebase/app'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase
      .firestore()
      .collection('recipes')
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}
