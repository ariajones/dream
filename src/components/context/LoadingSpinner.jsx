import React from 'react'

export default function LoadingSpinner () {
  return (
    <>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
    </>
  )
}