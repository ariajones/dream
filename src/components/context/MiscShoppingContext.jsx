import React, { useState, createContext, useEffect, useContext } from 'react'

import firebase from 'firebase/app'

import { UserContext } from './UserContext'

export const MiscShoppingContext = createContext()

export const MiscShoppingProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [miscCollection, setMiscCollection] = useState([])

  useEffect(() => {
    const unsubscribe = firebase // note unsubscribe added in case funny behaviour
      .firestore()
      .collection('miscShoppingList')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const miscCollection = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setMiscCollection(miscCollection)
      })

    return () => unsubscribe() // note unsubscribe added in case funny behaviour
  }, [user])

  return (
    <MiscShoppingContext.Provider value={[miscCollection, setMiscCollection]}>
      {children}
    </MiscShoppingContext.Provider>
  )
}
