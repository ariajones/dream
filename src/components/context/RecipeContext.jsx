import React, { useState, createContext, useEffect, useContext } from 'react'
import firebase from 'firebase/app'
import 'firebase/auth'

import { UserContext } from './UserContext'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase 
      .firestore()
      .collection('recipes')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [user])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}