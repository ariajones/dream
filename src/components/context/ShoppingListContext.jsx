import React, { useState, createContext, useEffect } from 'react'

import firebase from 'firebase/app'

export const ShoppingListContext = createContext()

export const ShoppingListProvider = ({ children }) => {
  const [shoppingList, setShoppingList] = useState([])

  useEffect(() => {
    const unsubscribe = firebase // note unsubscribe added in case funny behaviour
      .firestore()
      .collection('shoppingList')
      .onSnapshot(snapshot => {
        const newShoppingList = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setShoppingList(newShoppingList)
      })

    return () => unsubscribe() // note unsubscribe added in case funny behaviour
  }, [])

  return (
    <ShoppingListContext.Provider value={[shoppingList, setShoppingList]}>
      {children}
    </ShoppingListContext.Provider>
  )
}
