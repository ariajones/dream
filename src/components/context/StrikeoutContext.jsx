import React, { useState, createContext } from 'react'

export const StrikeOutContext = createContext()

export const StrikeOutProvider = ({ children }) => {
  const [striked, setStriked] = useState()

  return (
    <StrikeOutContext.Provider value={[striked, setStriked]}>
      {children}
    </StrikeOutContext.Provider>
  )
}