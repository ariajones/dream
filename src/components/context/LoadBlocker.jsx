import React, { useContext} from 'react'

import { RecipeContext } from './RecipeContext'
import { SelectedDayContext } from './SelectedDayContext'
import { ShoppingListContext } from './ShoppingListContext'
import { StrikeOut } from './StrikeoutContext'
import { UserContext } from './UserContext'
import { WeekContext } from './WeekContext'
import LoadingSpinner from './LoadingSpinner'


export default function LoaderBlocker ({ children }) {
  const {user} = useContext(UserContext)
  const [recipes] = useContext(RecipeContext)
  const [selectedDay] = useContext(SelectedDayContext)
  const [shoppingList] = useContext(ShoppingListContext)
  const [week] = useContext(WeekContext)
  // const [strikeOut] = useContext(StrikeOut)

  if (!user || !recipes || !selectedDay || !shoppingList || !week) {
    return <LoadingSpinner />
  } 
    return children
}