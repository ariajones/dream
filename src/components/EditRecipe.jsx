import React, { useState, useContext, useEffect } from 'react'
import firebase from 'firebase/app'

import { RecipeContext } from './context/RecipeContext'
import { updateRecipe, signOut } from '../utils'

export default function EditRecipe (props) {
  const [recipes] = useContext(RecipeContext)
  const recipeId = props.match.params.id
  const recipe = recipes.find(x => x.id === recipeId)

  const [imagePath, setImagePath] = useState(null)
  const [name, setName] = useState(null)
  const [serves, setServes] = useState(null)
  const [prepTime, setPrepTime] = useState(null)
  const [ingredients, setIngredients] = useState(null)
  const [method, setMethod] = useState(null)
  // const listedIngredients = ingredients ? ingredients.map(ingredient => {ingredient.quantity}{' '}{ingredient.name}</>) : null

  useEffect(() => {
    setImagePath(recipe ? recipe.imagePath : null)
    setName(recipe ? recipe.name : null)
    setServes(recipe ? recipe.serves : null)
    setPrepTime(recipe ? recipe.prepTime : null)
    setIngredients(recipe ? recipe.ingredients : null)
    setMethod(recipe ? recipe.method : null)
  }, [recipe])

  function onSubmitHandler (e) {
    e.preventDefault()
    const updatedRecipe = {
      imagePath,
      name,
      serves,
      prepTime,
      ingredients: ingredients,
      method: method
    }

    firebase
      .firestore()
      .collection('recipes')
      .doc(recipe.id)
      .update(updatedRecipe)
      .then(() => {
        setImagePath('')
        setName('')
        setServes('')
        setPrepTime('')
        setIngredients([])
        setMethod([])
      })
    props.history.push(`/recipe/${recipeId}`)
  }

  const clickHandler = () => {
    signOut()
    props.history.push('/')
  }

  return (
    <>
      <div className='noBulmaNav'>
        <img className="noBulmaNavLogo" src="../rp.png" alt="Logo"/>
        <div className='noBulmaNavTitle'>Edit Recipe</div>
        <h1 className="noBulmaNavSignOut" onClick={() => clickHandler()}>Sign Out</h1>
      </div>
      <form onSubmit={onSubmitHandler}>
        {/* <h4>Edit Recipe</h4> */}

        <div>
        <div className="card card-expanded-edit">
          <div className="card-image">
            <figure className="image is-2by1">
              <img src={imagePath} alt="add your meal" />
            </figure>
          </div>
          <div className="return-edit">
          <span><button className="button is-small is-rounded is-primary">
            <i className="fas fa-arrow-circle-left"></i>
          </button></span>
          </div>
          <div className="card-content">
            <div className="media">
              <div className="media-left">
                <figure className="image is-48x48">
                </figure>
              </div>
              <div className="media-content">
                <p className="subtitle is-5">
                  <div className="edit-title">
                    <label>Title</label>
                    <input className="label is-primary" type='text' value={name} onChange={e => setName(e.currentTarget.value)} /> {/* --- NAME OF RECIPE --- */}
                  </div>
                </p>
              </div>
            </div>
          </div>
          <div className="edit-image">
            <label>Image URL</label><br></br>
            <input type='text' value={imagePath} onChange={e => setImagePath(e.currentTarget.value)} />
          </div>
          <div className="content">
            <div className="card-content">
              Serves<br></br>
              <input value={serves} onChange={e => setServes(e.currentTarget.value)} /> {/* --- SERVES --- */}
            </div>
            <div className="card-content">
              Prep Time (minutes)<br></br>
              <input type='text' value={prepTime} onChange={e => setPrepTime(e.currentTarget.value)} /> {/* --- PREP TIME --- */}
            </div>
          </div>
          {/* <div className="card-content">
                Ingredients needed:
                (seperate by comma)
            <div>
              <textarea className="textarea is-primary" type='text' value={ingredients ? ingredients.map(ingredient => <p key={ingredient.name}>{ingredient.quantity}{' '}{ingredient.name}</p>) : null} onChange={e => setIngredients(e.currentTarget.value.split(','))} /> 
              <ul>{ingredients ? ingredients.map(ingredient => <li key={ingredient.id}>{ingredient.quantity}{' '}{ingredient.name}</li>) : null}</ul>
            </div>
          </div> */}
          
          <div className="">
            <div className="textmeth">
              <label>{'Method (seperate by comma)'}</label>
              <div className="editcard">
              <textarea className="textarea is-primary" type='text' value={method} onChange={e => setMethod(e.currentTarget.value.split(','))} />
              </div>
              <div className="methpad-edit">
              <ol>{method ? method.map(step => <li key={step}>{step}<br /><br /></li>) : null}
              </ol>
              </div>
            </div>
            <br></br>
            <div className="edit-button">
            <button className="button card-content is-small is-rounded is-primary edit-button">Edit Recipe</button>
            </div>
            </div>
          </div>
        </div>
        <br></br>
      </form>
    </>
  )
}

