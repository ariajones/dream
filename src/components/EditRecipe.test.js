import React, { useState } from 'react'
import renderer from 'react-test-renderer'

import EditRecipe from './EditRecipe'
import { RecipeContext } from './context/RecipeContext'

export const FakeRecipeProvider = ({ children }) => {
    const [recipes, setRecipes] = useState([{
    id: "MOqgTKsRp9OwepdSupbr",
    method: ["Preheat the oven to 200ºC/400ºF/gas 6. Peel 1 oni…"],
    serves: "4",
    prepTime: "40 Minutes - Not too tricky",
    name: "Roasted black bean burgers",
    ingredients: ["1½ red onions", "100 g rye bread", "ground corian…"],
    imagePath: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
    },
    {
        id: "29WXPJYg33PKdu0gAsTv",
    method: ["Preheat the oven to 190°C/375°F/gas 5.", "Melt 50…"],
    serves: "6",
    prepTime: "50 Minutes - Not too tricky",
    name: "Spinach lasagne",
    ingredients: ["70 g unsalted butter", "50 g plain flour", "800 m…"],
    imagePath: "https://img.jamieoliver.com/jamieoliver/recipe-database/2ipm2HQHql2Azq2dLfiA4k.jpg?tr=w-800,h-1066"
    }
])
  
    return (
        <RecipeContext.Provider value={[recipes, setRecipes]}>
          {children}
        </RecipeContext.Provider>
      )
    }
    
it ('renders component correctly', () => {
    const fakeParams = { params: {id: 'MOqgTKsRp9OwepdSupbr'} }
   
    const blah = renderer
        .create(
            <FakeRecipeProvider>
                <EditRecipe match={fakeParams}/>
            </FakeRecipeProvider>)
        .toJSON()
    expect(blah).toMatchSnapshot()
})

it ('renders component correctly', () => {
    const fakeParams = { params: {id: '29WXPJYg33PKdu0gAsTv'} }
   
    const blah = renderer
        .create(
            <FakeRecipeProvider>
                <EditRecipe match={fakeParams}/>
            </FakeRecipeProvider>)
        .toJSON()
    expect(blah).toMatchSnapshot()
})
