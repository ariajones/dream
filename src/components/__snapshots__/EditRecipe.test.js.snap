// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h4>
    Edit Recipe
  </h4>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <figure
        className="image is-2by1"
      >
        <img
          alt="add your meal"
          src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
        />
      </figure>
    </div>
    <span>
      <button
        className="button is-link is-small is-rounded"
      >
        <i
          className="fas fa-arrow-circle-left"
        />
      </button>
    </span>
    <div
      className="card-content"
    >
      <div
        className="media"
      >
        <div
          className="media-left"
        >
          <figure
            className="image is-48x48"
          />
        </div>
        <div
          className="media-content"
        >
          <p
            className="title is-5"
          >
            <div>
              <label>
                Title
              </label>
              <input
                className="label is-primary"
                onChange={[Function]}
                type="text"
                value="Roasted black bean burgers"
              />
               
            </div>
          </p>
        </div>
      </div>
    </div>
    <div>
      <label>
        Image URL
      </label>
      <input
        onChange={[Function]}
        type="text"
        value="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="card-content"
      >
        Serves
        <br />
        <input
          onChange={[Function]}
          value="4"
        />
         
      </div>
      <div
        className="card-content"
      >
        Prep Time
        <br />
        <input
          onChange={[Function]}
          type="text"
          value="40 Minutes - Not too tricky"
        />
         
      </div>
    </div>
    <div
      className="card-content"
    >
      Ingredients needed: (seperate by comma)
      <div>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "1½ red onions",
              "100 g rye bread",
              "ground corian…",
            ]
          }
        />
         
        <ul>
          <li>
            1½ red onions
          </li>
          <li>
            100 g rye bread
          </li>
          <li>
            ground corian…
          </li>
        </ul>
      </div>
    </div>
    <div
      className="card-content"
    >
      <div>
        <label>
          Method (seperate by comma)
        </label>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "Preheat the oven to 200ºC/400ºF/gas 6. Peel 1 oni…",
            ]
          }
        />
        <ol>
          <li>
            Preheat the oven to 200ºC/400ºF/gas 6. Peel 1 oni…
          </li>
        </ol>
      </div>
      <br />
      <button
        className="button card-content is-medium is-rounded is-primary"
      >
        Edit Recipe
      </button>
    </div>
  </div>
  <br />
</form>
`;

exports[`renders component correctly 2`] = `
<form
  onSubmit={[Function]}
>
  <h4>
    Edit Recipe
  </h4>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <figure
        className="image is-2by1"
      >
        <img
          alt="add your meal"
          src="https://img.jamieoliver.com/jamieoliver/recipe-database/2ipm2HQHql2Azq2dLfiA4k.jpg?tr=w-800,h-1066"
        />
      </figure>
    </div>
    <span>
      <button
        className="button is-link is-small is-rounded"
      >
        <i
          className="fas fa-arrow-circle-left"
        />
      </button>
    </span>
    <div
      className="card-content"
    >
      <div
        className="media"
      >
        <div
          className="media-left"
        >
          <figure
            className="image is-48x48"
          />
        </div>
        <div
          className="media-content"
        >
          <p
            className="title is-5"
          >
            <div>
              <label>
                Title
              </label>
              <input
                className="label is-primary"
                onChange={[Function]}
                type="text"
                value="Spinach lasagne"
              />
               
            </div>
          </p>
        </div>
      </div>
    </div>
    <div>
      <label>
        Image URL
      </label>
      <input
        onChange={[Function]}
        type="text"
        value="https://img.jamieoliver.com/jamieoliver/recipe-database/2ipm2HQHql2Azq2dLfiA4k.jpg?tr=w-800,h-1066"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="card-content"
      >
        Serves
        <br />
        <input
          onChange={[Function]}
          value="6"
        />
         
      </div>
      <div
        className="card-content"
      >
        Prep Time
        <br />
        <input
          onChange={[Function]}
          type="text"
          value="50 Minutes - Not too tricky"
        />
         
      </div>
    </div>
    <div
      className="card-content"
    >
      Ingredients needed: (seperate by comma)
      <div>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "70 g unsalted butter",
              "50 g plain flour",
              "800 m…",
            ]
          }
        />
         
        <ul>
          <li>
            70 g unsalted butter
          </li>
          <li>
            50 g plain flour
          </li>
          <li>
            800 m…
          </li>
        </ul>
      </div>
    </div>
    <div
      className="card-content"
    >
      <div>
        <label>
          Method (seperate by comma)
        </label>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={
            Array [
              "Preheat the oven to 190°C/375°F/gas 5.",
              "Melt 50…",
            ]
          }
        />
        <ol>
          <li>
            Preheat the oven to 190°C/375°F/gas 5.
          </li>
          <li>
            Melt 50…
          </li>
        </ol>
      </div>
      <br />
      <button
        className="button card-content is-medium is-rounded is-primary"
      >
        Edit Recipe
      </button>
    </div>
  </div>
  <br />
</form>
`;
